src/components/common/transactions-add/transactions-add.component.coffee

```diff
@@ -1,31 +1,78 @@
 module = angular.module('impac.components.common.transactions-add',[])
+                .constant('SCHEDULABLE_RULES', Object.freeze([
+                   {name: 'Daily', value: 'DailyRule'}, {name: 'Weekly', value: 'WeeklyRule'},
+                   {name: 'Monthly', value: 'Monthly'}, {name: 'Yearly', value: 'YearlyRule'}]))
 
 module.component('transactionsAdd', {
   templateUrl: 'common/transactions-add.tmpl.html'
   bindings:
     onHide: '&'
     onCreateTransaction: '&'
-    resourcesType: '='
-    contacts: '='
+    resourcesType: '<'
+    contacts: '<'
+    trx: '=?'
 
-  controller: ->
+  controller: (SCHEDULABLE_RULES)  ->
     ctrl = this
+    ctrl.schedulableRules = SCHEDULABLE_RULES
 
     ctrl.$onInit = ->
-      ctrl.trx =
-        datePicker: 
+      ctrl.editable = false
+
+      if(!ctrl.trx?)
+        ctrl.trx = {}
+        ctrl.editable = true
+
+      ctrl.trx.datePicker =
+        opened: false
+        date: new Date()
+        toggle: -> this.opened = !this.opened
+
+      ctrl.schedulable =
+        recurring: false
+        interval: 4
+        rule: ctrl.schedulableRules[2]
```

Comment:
Might be best to do `ctrl.schedulableRules.find((rule) -> rule.value == 'WeeklyRule')`, if a developer changes the order of that constant for some reason then this is broken. Also why is the `rule.value` in that capitalised camel case syntax and not something like `'weeklyrule'` or `'weekly_rule'` ?